backend/Database/user.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const userSchema = new Schema ({
+    firstName: {
+        type: String,
+        trim: true,
+        required: [ true, "name is required" ]
+    },
+    lastName: {
+        type: String,
+        trim: true
+    },
+    email: {
+        type: String,
+        unique: true,
+        trim: true,
+        required: [ true, "email is required" ],
+        validate: {
+            validator: function(v) {
+                return /[a-z][0-9]*@gmail.com/.test(v)
```

Comment:
doesn't this restrict the user to definitely have a gmail account?
This could be treated as a potential issue if someone wants to register using yahoo/outlook...